### DISCUSSION TITLE
Release 2024.0.0

### DISCUSSION THREAD

**Discussion by Nice3point** [jeremytammik/RevitLookup#149] [General]
## RevitLookup

Revit 2024 has been released, which means it's time to update RevitLookup
https://github.com/jeremytammik/RevitLookup/releases

## Also updated all additional components for developers:

**SDK**
https://www.autodesk.com/developer-network/platform-technologies/revit

**Libraries**
https://github.com/Nice3point/RevitApi

**Templates**
https://github.com/Nice3point/RevitTemplates

**Extensions**
https://github.com/Nice3point/RevitExtensions

**Comment by gaborschnierer** [comment]
Awesome job! Thank you for your contributions!
I really like the winui style wpf windows. I tried to achieve a similar result for my project, but failed. I'll dive into into the code, but if you can recommend any guides and tweaks to get a basic winui style window displayed in Revit, that would save me a ton of brainwork. 😅

**Comment by Nice3point** [comment]
Thank you, I'm very pleased) The easiest way is to copy the UI project from RevitLookup. You will not be able to use the original winui project because it is tied to the use of the static Appication.Current property, which is not available for the dll applications that are used in Revit.

In addition, you should not forget to add the manifest file to your application)
![изображение](https://user-images.githubusercontent.com/20504884/230770955-09e3ff7d-4b28-4c8f-b917-cb371c3a5d29.png)

There are probably no manuals, just learning the code 😉

Using Microsoft.Extensions.Hosting.Host is optional, you can create a regular window, new FluentWindow() and call the ShowDialog() method, if you tell me what fails you, I can give more information)

**Reply by gaborschnierer** [reply]
Thanks for the tip. Tried the Host way as well, but it just starts to get so dependent on RevitLookup classes like Host and IWindow, that I gave up. Alternatively newing up the FluentWindow and calling ShowDialog() throws when trying to remove the backdrop. Do you have any quick tips? If not, don't worry, it's not so vital.

**Reply by Nice3point** [reply]
The documentation used to be here  https://wpfui.lepo.co/documentation/, unfortunately it is not available nowTry running this project on your computer and learn how it works

**Reply by gaborschnierer** [reply]
Yea, I'll build it from ground up when I get the chance. Thanks! 🙏
https://github.com/lepoco/wpfui

**Comment by KPoeEVOLVE** [comment]
I miss being able to assign the "Current Selection" to a hotkey. Currently, I've bound the **Snoop Active view** to a hotkey, then I switch to the **Dashboard** and click **Selection**.  While it's only an extra two clicks, it's a pain nonetheless.

**Reply by Nice3point** [reply]
To be added in the next update. Release scheduled for tomorrow, stay tuned

**Reply by KPoeEVOLVE** [reply]
Awesome! Thank you :D
